youtube_to_anki/utils.py

```diff
@@ -24,3 +26,27 @@ def process_audio_chunk(audio: AudioSegment, chunk: Dict) -> AudioSegment:
     Indexes an audio segment by the start and end timestamp provided in the transcript chunk.
     """
     return audio[chunk["start"] : chunk["end"]]
+
+def take_screenshots(video: cv2.VideoCapture, chunks: Tuple[Dict]) -> Iterable[ndarray]:
+    """
+    Processes a video, for each interval taking a screenshot (ndarray, colors last) at around the middle of the interval
+    """
+    screenshots = []
+    # 30 ms offset to account for delay until the next frame
```

Comment:
for efficiency I just parse through all the "frames" in the video, which have a timestamp. If we just strictly look for a timestamp `>=` the expected one, then on average it will be delayed by ~30 ms or so, so I offset that delay